app/features/notes_generator/core.py

```diff
@@ -0,0 +1,69 @@
+import re
+from typing import Literal, Optional, Dict
+import logging
+from tools import (
+    extract_text_from_file,
+    extract_text_from_url,
+    generate_notes,
+    export_notes
+)
+
+def executor(input_type: Literal['text', 'file', 'url'],
+             input_content: Optional[str] = None,
+             file_path: Optional[str] = None,
+             focus_topic: Optional[str] = None,
+             output_structure: Literal['bullet', 'paragraph', 'table'] = 'bullet',
+             export_format: Literal['txt', 'docx', 'pdf'] = 'txt') -> Dict[str, str]:
```

Comment:
The standardize "export format" that the Marvel AI Backend supports is the JSON format for our UI client.